ext_libs/ext_libs.cmake

```diff
@@ -118,3 +118,14 @@ else()
   # header at the root of the repo rather than its own nested sse2neon/ dir
   target_include_directories(sse2neon SYSTEM INTERFACE "${CMAKE_CURRENT_LIST_DIR}/sse2neon")
 endif()
+
+add_library(mlpack_ensmallen INTERFACE)
+# ensmallen can only be installed from source so doesn't really make sense to add it as a system dependency
```

Comment:
All of our dependencies need to be consumable as a system dependency for things like package managers